### The Titanic Problem Notebook

In [45]:
import pandas as pd
import numpy as np
import re

titanic_data = pd.read_csv(r'..\titanic.csv')
titanic_data

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

### Beobachtungen und Fragen

Der Datensatz umfasst 12 Spalten und 891 Spalten mit Informationen zu Passagieren der Titanic. Anbei eine Übersicht der Spalten und der hinterlegten Informationen:
1. 'PassengerId' - enthält die PassagierId. Die hinterlegten Zahlen stehen für die IDs und reichen von 1 - 891.

2. 'Survived' - enthält den Überlebensstatus. 0 = hat nicht überlebt. 1 = hat überlebt.
 
3. 'Pclass' - Pclass enthält die Passagierklasse. Die Klassen reichen von 1 - 3.
Frage: Wo genau lagen die Kabinen der Klassen auf dem Schiff? Sind sie etwa Etagen zugeordnet? Wie ist der Zusammenhang zwischen Pclass und Parch?
  
4. 'Name' - enthält die nachfolgenden Informationen in der Reihenfolge:
                                                                        Nachname
                                                                        ,
                                                                        <Anrede/ Titel>, Namen

5. 'Sex'
'Age'
'SibSp',
'Parch'
'Ticket'
'Fare'
'Cabin'
'Embarked'


In [37]:
titanic_data.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

### 1. Trennung der Namensspalte

In [38]:
titanic_data['Name'].head()

0                              Braund, Mr. Owen Harris
1    Cumings, Mrs. John Bradley (Florence Briggs Th...
2                               Heikkinen, Miss. Laina
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)
4                             Allen, Mr. William Henry
Name: Name, dtype: object

In [42]:
titanic_data[['Surname', 'Name']] = titanic_data['Name'].str.split(', ', n=1, expand=True)

TypeError: DataFrame.insert() missing 1 required positional argument: 'value'

In [49]:
names_df = titanic_data['Name'].str.split(', ', n=1, expand=True)
titanic_data.insert(3, 'Surname', names_df[0])
titanic_data.insert(4, 'name', names_df[1])

In [55]:
titanic_data

PassengerId  Survived  Pclass    Surname  \
0              1         0       3     Braund   
1              2         1       1    Cumings   
2              3         1       3  Heikkinen   
3              4         1       1   Futrelle   
4              5         0       3      Allen   
..           ...       ...     ...        ...   
886          887         0       2   Montvila   
887          888         1       1     Graham   
888          889         0       3   Johnston   
889          890         1       1       Behr   
890          891         0       3     Dooley   

                                           name     Sex   Age  SibSp  Parch  \
0                               Mr. Owen Harris    male  22.0      1      0   
1    Mrs. John Bradley (Florence Briggs Thayer)  female  38.0      1      0   
2                                   Miss. Laina  female  26.0      0      0   
3            Mrs. Jacques Heath (Lily May Peel)  female  35.0      1      0   
4                             Mr. William Henry    male  35.0      0      0   
..                                          ...     ...   ...    ...    ...   
886                                 Rev. Juozas    male  27.0      0      0   
887                        Miss. Margaret Edith  female  19.0      0      0   
888              Miss. Catherine Helen "Carrie"  female   NaN      1      2   
889                             Mr. Karl Howell    male  26.0      0      0   
890                                 Mr. Patrick    male  32.0      0      0   

               Ticket     Fare Cabin Embarked  
0           A/5 21171   7.2500   NaN        S  
1            PC 17599  71.2833   C85        C  
2    STON/O2. 3101282   7.9250   NaN        S  
3              113803  53.1000  C123        S  
4              373450   8.0500   NaN        S  
..                ...      ...   ...      ...  
886            211536  13.0000   NaN        S  
887            112053  30.0000   B42        S  
888        W./C. 6607  23.4500   NaN        S  
889            111369  30.0000  C148        C  
890            370376   7.7500   NaN        Q  

[891 rows x 13 columns]